<?xml version="1.0"?>
<launch>
  <arg default="true" doc="If true, limits joint range [-PI, PI] on all joints." name="limited"/>
  <arg default="false" doc="Starts gazebo in paused mode" name="paused"/>
  <arg default="true" doc="Starts gazebo gui" name="gui"/>
  <arg default="true" doc="If True, simulated instance of UR5e" name="sim"/>
  <arg default="false" doc="If True, MoveIt will run in debugging mode" name="debug"/>
  <arg default="left_" doc="The left (from base link reference frame) robot's name prefix" name="left_prefix"/>
  <arg default="right_" doc="The right (from base link reference frame) robot's name prefix" name="right_prefix"/>
  <arg default="false" doc="If True, MoveIt will run in debugging mode" name="moveit_debug"/>

  <!-- load up params to param server -->
  <rosparam command="load" file="$(find ur5e_2f85)/config/ur5e_2f85_params.yaml"/>

  <!-- startup simulated world -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg default="worlds/empty.world" name="world_name"/>
    <arg name="paused" value="$(arg paused)"/>
    <arg name="gui" value="$(arg gui)"/>
  </include>
  
  <!-- Spawn a table at the origin, slightly shifted down in Z-->
  <!--
  <node name="spawn_table" pkg="gazebo_ros" type="spawn_model" args="-file /home/james/.gazebo/models/cafe_table/model.sdf -sdf -x 1.0 -y 0.0 -z -.4 -model table"/>
  -->
  
  <!-- Send robot urdf to param server -->
  <node args="--robot_side single --limited True" name="upload_ur5e_2f85" output="screen" pkg="ur5e_2f85" type="upload_ur5e_2f85.py" respawn="false"/>

  <!-- Launch Rviz w/ single configuration -->
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find uncalibrated_grasping)/config/single_ur5e_2f85.rviz"/>  

</launch>